From 5cf52dd381fbae09f94f860932a77237e85445fc Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 24 Jul 2014 18:16:04 -0400
Subject: [fs] nfs: check the return of nfs4_negotiate_security in nfs4_submount

Message-id: <1406225773-16395-2-git-send-email-steved@redhat.com>
Patchwork-id: 86633
O-Subject: [RHEL7.1 PATCH 01/10] NFS check the return of nfs4_negotiate_security in nfs4_submount
Bugzilla: 1120728
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Andy Adamson <andros@netapp.com>

Signed-off-by: Andy Adamson <andros@netapp.com>
Tested-By: Steve Dickson <steved@redhat.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 57bbe3d7c110fcc20a661cb814a401e4126ca590
Bugzilla: 1120728

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4namespace.c b/fs/nfs/nfs4namespace.c
index 3d5dbf8..1b86fef 100644
--- a/fs/nfs/nfs4namespace.c
+++ b/fs/nfs/nfs4namespace.c
@@ -399,8 +399,11 @@ struct vfsmount *nfs4_submount(struct nfs_server *server, struct dentry *dentry,
   flavor = client->cl_auth->au_flavor;
  else {
   rpc_authflavor_t new = nfs4_negotiate_security(dir, name);
-  if ((int)new >= 0)
-   flavor = new;
+  if ((int)new < 0) {
+   mnt = ERR_PTR((int)new);
+   goto out;
+  }
+  flavor = new;
  }
  mnt = nfs_do_submount(dentry, fh, fattr, flavor);
 out:
-- 
1.7.1